pkg/driver/controller_server.go

```diff
@@ -622,8 +622,84 @@ func (d *Driver) DeleteSnapshot(ctx context.Context, req *csi.DeleteSnapshotRequ
 	return nil, status.Error(codes.Unimplemented, "")
 }
 
-// ListSnapshots is part of implementing Snapshot & Restore functionality, but we don't support that
-func (d *Driver) ListSnapshots(context.Context, *csi.ListSnapshotsRequest) (*csi.ListSnapshotsResponse, error) {
+// ListSnapshots retrieves a list of existing snapshots as part of the Snapshot & Restore functionality.
+func (d *Driver) ListSnapshots(ctx context.Context, req *csi.ListSnapshotsRequest) (*csi.ListSnapshotsResponse, error) {
+	log.Info().Msg("Request: ListSnapshots")
+
+	snapshotID := req.GetSnapshotId()
+	sourceVolumeID := req.GetSourceVolumeId()
+
+	if req.GetStartingToken() != "" {
+		log.Error().Msgf("Pagination not supported")
+		return nil, status.Errorf(codes.Aborted, "%v not supported", "starting-token")
+	}
+
+	// case 1: SnapshotId is not empty, return snapshots that match the snapshot id
+	if len(snapshotID) != 0 {
+		log.Debug().Msgf("Fetching snapshot with ID %v", snapshotID)
+
+		// Retrieve a specific snapshot by ID
+		// Todo: GetSnapshot to be implemented in civogo
+		// Todo: Un-comment post client implementation
+		// snapshot, err := d.CivoClient.GetSnapshot(snapshotID)
+		// if err != nil {
+		// 	log.Error().Err(err).Msgf("No snapshot found matching the ID %v", snapshotID)
+		// 	if strings.Contains(err.Error(), "DatabaseSnapshotNotFoundError") {
+		// 		return &csi.ListSnapshotsResponse{}, nil
+		// 	}
+		// 	return nil, status.Errorf(codes.Internal, "failed to list snapshot %v: %v", snapshotID, err)
```

Comment:
I think it would be easier to debug if the log were changed according to error. What do you think? :thinking: 
```suggestion
		// 	if strings.Contains(err.Error(), "DatabaseSnapshotNotFoundError") {
		// 		log.Info().
		// 			Str("snapshot_id", snapshotID).
		// 			Msg("No snapshot found")
		// 		return &csi.ListSnapshotsResponse{}, nil
		// 	}
		// 	log.Error().
		// 		Str("snapshot_id", snapshotID).
		// 		Err(err).Msg("Failed to get snapshot from Civo API")
		// 	return nil, status.Errorf(codes.Internal, "failed to list snapshot %q: %v", snapshotID, err)
```

Reply:
//    if strings.Contains(err.Error(), "DatabaseSnapshotNotFoundError") {  
        // 		log.Error().
        //          Err(err).
        // 			Str("snapshot_id", snapshotID).  
        // 			Msg("No snapshot found")  
        // 		return &csi.ListSnapshotsResponse{}, nil  
        // 	}  

        Can this be an 'error' instead of 'info', so we can log the actual error received from the API as well along with the message?
```suggestion
		// snapshot, err := d.CivoClient.GetSnapshot(snapshotID)
		// if err != nil {
		// 	if strings.Contains(err.Error(), "DatabaseSnapshotNotFoundError") {
		// 		log.Info().  
        // 			Str("snapshot_id", snapshotID).  
```